### PR TITLE
adds support of CodeS3Location in AppSyncFunctionConfiguration

### PR DESCRIPTION
**Pull request by roh-hitiq**
#### Which issue(s) does this change fix?

New AppSync Javascript resolvers allow you to write resolvers in JS rather than VTL.
Currently when you add a local reference for a CodeS3Location it is not resolved by the formatter
#### Why is this change necessary?
Currently cannot include Javascript resolves as part of 

#### How does it address the issue?
Adds  `CodeS3Location` to AWS_APPSYNC_FUNCTIONCONFIGURATION in resources.
Adds `AppSyncFunctionConfigurationCodeResource` to Package resources

#### What side effects does this change have?
None

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Comment by jfuss** [comment]
@roh-hitiq In the future, please follow our [Contributing Guidelines](https://github.com/aws/aws-sam-cli/blob/develop/CONTRIBUTING.md#contributing-via-pull-requests) and submit an Issue first. 

Seems reasonable change but needs tests. Can you please add them?

**Review by jfuss — APPROVED** [review]
Looks reasonable to me. Thanks for updating with tests.

**Review by sriram-mv — APPROVED** [review]

**Comment by jfuss** [comment]
@roh-hitiq I thought I commented this before, so my apologizes. Could you run `make black`. It appears the ci is failing due to formatting issues.

**Comment by jfuss** [comment]
This solves #4482

File: samcli/lib/utils/resources.py [hunk 1]
```diff
@@ -59,7 +59,7 @@
     AWS_SERVERLESS_STATEMACHINE: ["DefinitionUri"],
     AWS_APPSYNC_GRAPHQLSCHEMA: ["DefinitionS3Location"],
     AWS_APPSYNC_RESOLVER: ["RequestMappingTemplateS3Location", "ResponseMappingTemplateS3Location"],
```
Comment (jfuss) [inline samcli/lib/utils/resources.py:61]: @roh-hitiq Can you add `CodeS3Location` to this list as well?

Comment (roh-hitiq) [reply samcli/lib/utils/resources.py]: This has been added

**Review by mndeveci — APPROVED** [review]

**Pull request merged**